vinyldns/integration_test.go

```diff
@@ -397,6 +397,59 @@ func TestRecordSetsListAllIntegrationFilterForNonexistentName(t *testing.T) {
 	}
 }
 
+func TestRecordSetsGlobalListAllIntegrationFilterForExistentName(t *testing.T) {
+	c := client()
+	rName := "global-list-int-test"
+	zs, err := c.ZonesListAll(ListFilter{})
+	if err != nil {
+		t.Error(err)
+	}
+
+	_, err = c.RecordSetCreate(&RecordSet{
+		Name:   rName,
+		ZoneID: zs[0].ID,
+		Type:   "A",
+		TTL:    60,
+		Records: []Record{
+			{
+				Address: "127.0.0.1",
+			},
+		},
+	})
+	if err != nil {
+		t.Error(err)
+	}
+
+	records, err := c.RecordSetsGlobalListAll(GlobalListFilter{
+		RecordNameFilter: rName,
```

Comment:
You need either the FQDN here, or a wildcard
```suggestion
		RecordNameFilter: rName + "*",
```